core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -131,14 +132,15 @@ private static Map<String, String> loadEventListenerProperties(File configFile)
         }
     }
 
-    public void queryCompleted(QueryCompletedEvent queryCompletedEvent)
+    public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventSupplier)
```

Comment:
Supplier typically means a getter without args, `queryCompletedEventProvider` might be better name